server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_new_scheduled_publish(self, mock_factory, mock_reverse, mock_resp, mock_redir):
+        """
+        Create a new scheduled publish.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({})
+        mock_schedule = mock.MagicMock()
+        mock_schedule.for_display.return_value = {'_id': 'mock_schedule'}
+        mock_factory.repo_publish_schedule_manager.return_value.create.return_value = mock_schedule
+        mock_reverse.return_value = '/mock/path/'
+
+        publish_schedule = RepoPublishSchedulesView()
+        response = publish_schedule.post(mock_request, 'mock_repo', 'mock_distributor')
+
+        mock_factory.repo_publish_schedule_manager.return_value.create.assert_called_once_with(
+            'mock_repo', 'mock_distributor', {'override_config': {}}, None, None, True
+        )
+        mock_resp.assert_called_once_with({'_id': 'mock_schedule', '_href': '/mock/path/'})
+        self.assertTrue(response is mock_redir.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_CREATE())
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_post_new_scheduled_publish_extra_fields(self, mock_factory):
+        """
+        Create a new scheduled publish with extra nonallowed fields.
+        """
+
+        mock_request = mock.MagicMock()
+        mock_request.body = json.dumps({'schedule': 'some iso8601', 'extrafield': 'cause failure'})
+        publish_schedule = RepoPublishSchedulesView()
+        try:
+            publish_schedule.post(mock_request, 'mock_repo', 'mock_distributor')
+        except pulp_exceptions.UnsupportedValue, response:
+            pass
+        else:
+            raise AttributeError("UnsupportedValue should be raised with extra fields in body")
+
+        self.assertEqual(response.http_status_code, 400)
+        self.assertTrue(response.error_code is error_codes.PLP0017)
+
+
+class TestRepoPublishScheduleResourceView(unittest.TestCase):
+    """
+    Tests for the RepoPublishScheduleResourceView.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch('pulp.server.webservices.views.repositories.RepoPublishScheduleResourceView._get')
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.manager_factory.repo_publish_schedule_manager')
+    @mock.patch('pulp.server.webservices.views.repositories.reverse')
+    def test_get_publish_schedule(self, mock_rev, mock_manager, mock_get):
+        """
+        Test retrieval of a single scheduled publis
```

Comment:
i think you can get rid of mock_get_repo() just by doing:
mock_factory.repo_query_manager.return_value.get_repository.side_effect = MissingResource()